protocol/hash.go

```diff
@@ -33,6 +33,7 @@ const (
 	BalanceRecord     HashID = "BR"
 	Credential        HashID = "CR"
 	Genesis           HashID = "GE"
+	Logic             HashID = "LO"
 	Message           HashID = "MX"
```

Comment:
This isn't strictly part of this PR, but could you move `multiSigString` from `crypto/multisig.go` into this list of `HashID` values?  Now that we have other things being hashed into addresses (specifically, these new logic addresses), it's important that the hash input for multisig addrs is domain-separated from logic addresses.

Reply:
If multisig.go was generating the multisig Address values through the cypto.Hashable interface that would make sense, but I think because multisig.go is doing its own custom byte appending to build the value that gets hashed I'd rather leave that alone.

Reply:
It would be good to convert it to use `crypto.Hashable`; it's too bad it wasn't written using `crypto.Hashable` to begin with.  The reason it's related to this PR is that, with this PR, it becomes important that multisig and logicsig are domain-separated inputs to our hash function.

Reply:
Nitpick: Can we start making our domain separation prefixes longer/more meaningful than two bytes? e.g., "LogicAddress" instead of "LO". They're never transmitted over the wire so we don't need to make them small.
(See also my comment in https://github.com/algorand/go-algorand/issues/204)